### PR TITLE
MNT: simplify AutoDateFormatter

### PR DESCRIPTION
**Pull request by tacaswell**
- import constants from matplolib.dates
- do not re-implement AutoDateFormatter functionality that
  is already upstream

### CODE REVIEW

**Comment by jreback** [comment]
can you show an example of where this was doing the wrong thing? (e.g. a test for this)

**Comment by tacaswell** [comment]
There is nothing wrong, it just was an almost copy-paste job from mpl to pandas.  This change should have 0 functional change.

**Comment by jreback** [comment]
ok, seems like we were just overriding mpl code. Since everything passed, this either was never tested (probably), or works ok :)

File: pandas/tseries/converter.py [hunk 1]
```diff
@@ -222,31 +220,6 @@ def __init__(self, locator, tz=None, defaultfmt='%Y-%m-%d'):
         # matplotlib.dates._UTC has no _utcoffset called by pandas
         if self._tz is dates.UTC:
             self._tz._utcoffset = self._tz.utcoffset(None)
-        self.scaled = {
-            365.0: '%Y',
-            30.: '%b %Y',
-            1.0: '%b %d %Y',
-            1. / 24.: '%H:%M:%S',
-            1. / 24. / 3600. / 1000.: '%H:%M:%S.%f'
```
Comment (tacaswell) [inline pandas/tseries/converter.py]: Found the difference between what was here and what is in MPL it is the extra 1000 in the last entry.

I will have a PR shortly.

**Pull request merged**